### PR TITLE
[BUGFIX] Pass in partitioner + batching_regex when creating batch_def…

### PR DESCRIPTION
**Pull request by tyler-hoffman**
…inition

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/datasource/fluent/interfaces.py [hunk 1]
```diff
@@ -1085,7 +1085,9 @@ def _validator(self) -> V1Validator:
                 "We can't validate batches that are attached to datasources without a data context"
             )
         batch_definition = self.data_asset.add_batch_definition(
```
Comment (tyler-hoffman) [inline great_expectations/datasource/fluent/interfaces.py:1086]: I'm stressed that we are adding a new batch_definition here. I want to put up a rival PR to see if we can solve the bug and also avoid this side-effect. Edit: rival PR is here: https://github.com/great-expectations/great_expectations/pull/9799

File: great_expectations/datasource/fluent/interfaces.py [hunk 2]
```diff
@@ -1085,7 +1085,9 @@ def _validator(self) -> V1Validator:
...
-            name="-".join([self.datasource.name, self.data_asset.name, str(uuid.uuid4())])
+            name="-".join([self.datasource.name, self.data_asset.name, str(uuid.uuid4())]),
+            partitioner=self.batch_request.partitioner,
+            batching_regex=self.batch_request.batching_regex,
```
Comment (billdirks) [inline great_expectations/datasource/fluent/interfaces.py]: I didn't realize the batching_regex was separate from the partitioner in the batch request.

**Review by billdirks — APPROVED** [review]
LGTM, thanks!

File: great_expectations/datasource/fluent/interfaces.py (hunk 2 above)
Comment (tyler-hoffman) [reply great_expectations/datasource/fluent/interfaces.py]: I think that will change with @joshua-stauffer 's work

Comment (tyler-hoffman) [reply great_expectations/datasource/fluent/interfaces.py]: fyi Josh's work has merged, so this PR is now updated to reflect that batching regex is just nested in sql partitioners

**Pull request merged**